### PR TITLE
[DCA][Flare] Adds `DatadogAgent` user values

### PR DESCRIPTION
**Pull request by tbavelier**
### What does this PR do?
* Same as https://github.com/DataDog/datadog-agent/pull/16313 but with the operator: adds redacted (using the flare redaction system)`DatadogAgent` user values by retrieving it from the API server
* Renames the environment variables that are not Helm specific

### Motivation
Same motivation, help with troubleshooting:
> As containers support team, I would like to get all the info needed directly from the flare. We usually ask customer to provide the helm value files and the daemonset|deployment manifests
As customer, I would like that my support ticket require the least possible roundtrip iterations to speedup the resolution.

### Describe how to test/QA your changes
Using a `kind` cluster with the Operator built from below PR:
1. Deploy a `DatadogAgent` manifest
2. Run the flare command in the DCA
3. Retrieve the archive, unzip it and assert the presence of:
    - [ ] `datadog-agent-cr.yaml`: the `DatadogAgent` resource
    - [ ] `agent-daemonset.yaml`: the node Agent daemonset manifest
    - [ ] `cluster-agent-deployment.yaml`: the DCA deployment manifest 
image

### Possible Drawbacks / Trade-offs
N/A

### Additional Notes
* This requires https://github.com/DataDog/datadog-operator/pull/1477 that adds the necessary env vars on the cluster Agent to function

### CODE REVIEW

File: releasenotes/notes/add-datadogagent-manifest-flare-d842872ee18fe0c1.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+enhancements:
+  - |
+    Cluster Agent: ``DatadogAgent`` custom resource, cluster Agent deployment and node Agent daemonset manifests are now added to the flare archive, when the Cluster Agent is deployed with the Datadog Operator (version 1.11.0+).
```
Comment (jhgilbert) [inline releasenotes/notes/add-datadogagent-manifest-flare-d842872ee18fe0c1.yaml:4]:
```suggestion
    Cluster Agent: ``DatadogAgent`` custom resource, cluster Agent deployment, and node Agent daemonset manifests are now added to the flare archive when the Cluster Agent is deployed with the Datadog Operator (version 1.11.0+).
```

**Review by jhgilbert — APPROVED** [review]
Approved with very minor grammar suggestion, thanks!

File: pkg/flare/manifests.go
Comment (pgimalac) [inline pkg/flare/manifests.go:1]: Could you add some codeowners on this file ?

**Review by pgimalac — APPROVED** [review]
Hard to review the code added in manifest as it's really out of my knowledge base, but LGTM

File: pkg/flare/manifests.go
Comment (tbavelier) [reply pkg/flare/manifests.go:1]: Added in 99055a06c210b50f61f9376244c82bb7a0669303

**Pull request merged**